src/pointerToPath.ts

```diff
@@ -0,0 +1,37 @@
+import { replaceInString } from './_utils';
+
+export const pointerToPath = (pointer: string): string[] => {
+  return decodeUriFragmentIdentifier(pointer);
+};
+
+const decodeFragmentSegments = (segments: string[]): string[] => {
+  const len = segments.length;
+  const res = new Array(len);
+  let i = -1;
+
+  while (++i < len) {
+    res[i] = replaceInString(replaceInString(decodeURIComponent('' + segments[i]), '~1', '/'), '~0', '~');
+  }
+
+  return res;
+};
+
+const decodeUriFragmentIdentifier = (ptr: string): string[] => {
+  if (typeof ptr !== 'string') {
+    throw new TypeError('Invalid type: JSON Pointers are represented as strings.');
+  }
+
+  if (ptr.length === 0 || ptr[0] !== '#') {
+    throw new ReferenceError('Invalid JSON Pointer syntax; URI fragment idetifiers must begin with a hash.');
```

Comment:
I like an idea of a different error classes for a different error types, but `ReferenceError` is mostly used when a non-existent variable is referenced. IMO `URIError` or a regular `Error` are more suitable in this case.